### DISCUSSION TITLE
Will joomla 3 component structure be deprecated?

### DISCUSSION THREAD

**Discussion by trananhmanh89** [joomla/joomla-cms#35482] [General]
Hi,
I'm just wondering, will joomla 3 component strucuture be deprecated? Can i use j3 folder structure for future j4, j5?

As i see, joomla 4 has a very different (and complex) way to create a component. 

Thanks

**Comment by Bakual** [comment]
If I have to guess: J5 almost certainly will require namespaced code (the "J4" way).
The current J3 structure however will certainly work during the whole J4 life. It may be that new features only support the namespaced code. But so far that is not the case to my knowledge.

As for J5: It may well be that the current J4 structure is outdated by then as well. Namespaces will certainly be still a thing, but who knows what PHP will bring in the years to come till J5 😄

**Comment by trananhmanh89** [comment]
@brianteeman could you tell me?

**Comment by PhocaCz** [comment]
This is in fact very important question. We should differentiate between MVC folder structure and namespaces used in classes/methods.

Transforming methods and classes to their namespaces alternative is OK and can be done per different automated scripts. In fact there should not be a problem to transform it during Joomla 4 lifecycle.

The problem is MVC folder structure which is completely different. We can say that now, maybe 95% or more of all extensions use Joomla 3 MVC. Such extensions can be updated to Joomla 4 more or less quickly. But if Joomla 3 MVC will be removed (e.g. in Joomla 5), a lot of extensions just won't be able to be overwritten for Joomla 4/5 MVC because mostly using of automated scripts will not help.

The conclusion may be that if Joomla 5 is released in the standard timeframe and Joomla 3 MVC is removed, then we can expect a massive loss of extensions.

**Comment by laoneo** [comment]
I guess that even J3 MVC structure will work to some degree in J5. What is more painful is that all the views have to be adopted on every Joomla major release. And this is what will cause the loss @PhocaCz is talking about. Because it is really time consuming to adapt the views on large components for every major. In this are we have to be more BC.
